src/BrowserDetector.js

```diff
@@ -1,9 +1,6 @@
-import memoize from 'lodash/memoize';
 
-export const isFirefox = memoize(() =>
-  /firefox/i.test(navigator.userAgent)
-);
+const _isSafari = Boolean(window.safari);
+const _isFirefox = !_isSafari && /firefox/i.test(navigator.userAgent);
 
-export const isSafari = memoize(() =>
-  Boolean(window.safari)
-);
+export const isFirefox = () => _isFirefox;
```

Comment:
Probably tried to avoid blowing up server rendering. Should be fine if you add checks for `navigator` / `window` and do this eagerly.